docs/src/pages/demos/autocomplete/IntegrationAutosuggest.js

```diff
@@ -76,11 +75,11 @@ function renderSuggestion(suggestion, { query, isHighlighted }) {
       <div>
         {parts.map((part, index) => {
           return part.highlight ? (
-            <span key={index} style={{ fontWeight: 300 }}>
+            <span key={String(index)} style={{ fontWeight: 300 }}>
```

Comment:
Nice trick.

Reply:
I’m curious what the point of this is... I use numbers for keys all the time in React, is it not advised?

Reply:
Technically the key should be unique so that React knows what each thing is in the DOM. If you end up with two items with the same key, then it is possible that React won't render the second one. That's an over-simplification, but you likely get the point. Not sure that forcing a number to be a string would get around that, though.

Reply:
It only needs to be unique within its parent, which using an increment index surely accomplishes...

Reply:
I thought React warned if you used an index as a key (for the reason @pelotom mentioned), and took this to be a way to avoid the warning when it's known the list won't change, but I don't see a wanring for this example in the docs (without `String`), so 🤷‍♂️.

Reply:
AFAIK you only get a warning if you leave the `key` out completely

Reply:
`String()` is a "hack" to dodge https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-array-index-key.md.

Reply:
Aha, it all becomes clear 😄

Reply:
Ah, right, it's an ESLint warning: 

```
error  Do not use Array index in keys  react/no-array-index-key
```

Reply:
Bah, you beat me to it @oliviertassinari. 😄